### DISCUSSION TITLE
Pass Function As Attrubute Value

### DISCUSSION THREAD

**Discussion by HeadMonitor** [AvaloniaUI/Avalonia#13481] [Q&A] [answered]
Hello, I am really new to Avalonia and I am not sure on how to achieve this. I have made a UserControl as a reusable template for a custom button.

```xml
<Button HorizontalAlignment="Stretch" Click="{Binding ButtonClick}">
        <Grid ColumnDefinitions="Auto, 5, Auto">
            <avalonia:MaterialIcon Grid.Column="0" Kind="{Binding IconKind}"/>
            <TextBlock Grid.Column="2" Text="{Binding ButtonText}"/>
        </Grid>
</Button>
```

As you can see I want the **Click** attribute to be passed on from whatever control I try to use this button. The backend code for this part is the following:

```cs
public static readonly StyledProperty<Func<string>> ButtonClickProperty =
          AvaloniaProperty.Register<IconButton, Func<string>>(nameof(ButtonClick));
            
public Func<string> ButtonClick
        {
            get => GetValue(ButtonClickProperty);
            set => SetValue(ButtonClickProperty, value);
        }            
```

I want it so that if I call it from a control named "Test.axaml" in Test I can pass the name of a function that is in "Test.axaml.cs".

I cannot get this to work. I have tried many things but with no progress. If you can help me I would highly appreciate it!

**Comment by workgroupengineering** [comment]
Use Command https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/MVVM/CommandSample

**Comment by timunie** [comment, marked as answer]
Why not inherit Button? `MyButton : Button`? Then you have all the properties and functions for free. Add the needed Style for MyButton (copy main parts form Button-stlye and you're done. 

https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CustomControls/RatingControlSample

**Reply by HeadMonitor** [reply]
Aah! I did not think of that at all! Thanks a lot, works like a charm!

**Reply by HeadMonitor** [reply]
It seems that I spoke a bit too soon because when I inherit Button the buttons don't render. Any idea what might be causing this?

**Reply by workgroupengineering** [reply]
```csharp
protected override Type StyleKeyOverride => typeof(YourControl);
```

**Reply by HeadMonitor** [reply]
That did the trick! Thanks a lot!!
